tests/testsuite/script.rs

```diff
@@ -507,9 +507,9 @@ fn main() {
             "\
 [WARNING] `package.edition` is unspecifiead, defaulting to `2021`
 [COMPILING] bar v0.0.1 ([ROOT]/foo/bar)
-[COMPILING] script v0.0.0 ([ROOT]/home/.cargo/eval/target/eval/[..]/script)
+[COMPILING] script v0.0.0 ([ROOT]/foo)
```

Comment:
If changing this cargo package from a `src/lib.rs` to a `src/main.rs`, an error occurs:

```
error: `cargo run` could not determine which binary to run. Use the `--bin` option to specify a binary, or the `default-run` manifest key.
available binaries: script, script_939dd8258a9d52c84c9e91a655f83a25d1c75ec1bcd0cfdfa08e2c9a8c501603
```

I think either the cargo target discovery needs to be turned off, or setting `default-run`, or both.

`cargo run` enumerate executables here to determine which to run

https://github.com/rust-lang/cargo/blob/fdc29046a24e0fef601983d8896f61c8e7d0d5b0/src/cargo/ops/cargo_run.rs#LL27C1-L27C1